src/WebWindow.Native/WebWindow.Windows.cpp

```diff
@@ -225,7 +225,18 @@ void WebWindow::AttachWebView()
 						Settings->put_IsScriptEnabled(TRUE);
 						Settings->put_AreDefaultScriptDialogsEnabled(TRUE);
 						Settings->put_IsWebMessageEnabled(TRUE);
-
+                        // Add a navigation change event handler
+						EventRegistrationToken token;
+                        _webviewWindow->add_NavigationStarting(Callback<IWebView2NavigationStartingEventHandler>(
+                            [this](IWebView2WebView* webview, IWebView2NavigationStartingEventArgs * args) -> HRESULT {
+                                PWSTR uri;
+                                args->get_Uri(&uri);
+                                _uriChangeCallback(uri);
+                                CoTaskMemFree(uri);
+                                return S_OK;
```

Comment:
The logic in this block doesn't look right to me. Elsewhere we use `wil::unique_cotaskmem_string` and don't need to call `CoTaskMemFree`. Could you see how the strings are handled elsewhere (e.g., in `add_WebResourceRequested`) and follow the same pattern?

Reply:
I pulled this from 

https://docs.microsoft.com/en-us/microsoft-edge/hosting/webview2/gettingstarted#step-4---navigation-events

I do not have a problem removing it, but if you could give me direction to which you would prefer, I can do so.